### PR TITLE
Closes #8264: Pass copy of breadcrumbs to crash reporting services

### LINKED ISSUE #8264: Pass copy of breadcrumbs to crash reporting services
Currently crash reporter is passing breadcrumbs to crash reporting services directly.  In some cases `ConcurrentModificationException` is seen where breadcrumbs are being modified while crash reporting services are looping through it.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt [hunk 1]
```diff
@@ -96,6 +96,13 @@ class CrashReporter(
         return this
     }
 
+    /**
+     * Get a copy of the crashBreadCrumbs
+     */
+    fun crashBreadcrumbsCopy(): ArrayList<Breadcrumb> {
+        return ArrayList<Breadcrumb>(crashBreadcrumbs)
+    }
```
Comment (pocmo) [inline components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt:104]: This shouldn't be a public method for everyone to call :)

(Afaik you can use `.clone()` for that)

Comment (rocketsroger) [reply components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt:104]: My bad.  Should have been internal.  I'll update.  Thanks,

**Comment by rocketsroger** [comment]
Latest test report https://sentry.prod.mozaws.net/operations/reference-browser/issues/9471214/events/50894721/

File: components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt (hunk 1 above)
Comment (rocketsroger) [reply components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt:105]: Updated to use clone as well 👍

**Review by pocmo — APPROVED** [review]

**Pull request merged**